File: CodeChef/UNITGCD Unit GCD.py
'''Problem Link - https://www.codechef.com/APRIL20B/problems/UNITGCD'''
from sys import stdin

def read(n,arr):
    if n > 1:
        print(n//2)
        if n&1:
            print(3,arr[0],arr[1],arr[-1])
            for i in range(2,n-2,2):
                print(2,arr[i],arr[i+1])
        else:
            for i in range(1,n,2):
                print(2,arr[i-1],arr[i])
    else:
        print(1)
        print(1,1)
for _ in range(int(stdin.readline())):
    n = int(stdin.readline())
    arr = list(range(1,n+1))
    read(n,arr)

File: CodeChef/UNITGCD_Unit_GCD.c
// https://www.codechef.com/APRIL20B/problems/UNITGCD   


#include <stdio.h>

int main(void) {
	// your code goes here
	int t;
	scanf("%d", &t);
	while(t>0){
		t--;
		long long int i;
		long long int n;
		scanf("%lld", &n);
		if(n==1){
			printf("1");
			printf("\n");
		}
		else
			printf("%lld\n", n/2);
		if (n <= 3) {
			printf("%lld ", n);
			for(i = 1;i<=n;i++){
				printf("%lld ", i);
			}
		}
		else {
			printf("3 1 2 3\n");
			for(i=4;i<n;i=i+2){
				printf("2 ");
				printf("%lld %lld\n", i,i+1);
			}
			if (n%2 == 0){
				printf("1 ");
				printf("%lld\n", n );
			}
			
		}
	}
	return 0;
}
